ossdbtoolsservice/object_explorer/object_explorer_service.py

```diff
@@ -206,18 +206,24 @@ def _expand_node_base(self, is_refresh: bool, request_context: RequestContext, p
                 session.refresh_tasks[key] = new_task
             else:
                 session.expand_tasks[key] = new_task
-
         except Exception as e:
             self._expand_node_error(request_context, params, str(e))
 
-    def _expand_node_thread(self, is_refresh: bool, request_context: RequestContext, params: ExpandParameters, session: ObjectExplorerSession):
+    def _expand_node_thread(self, is_refresh: bool, request_context: RequestContext, params: ExpandParameters, session: ObjectExplorerSession, retry=False):
         try:
             response = ExpandCompletedParameters(session.id, params.node_path)
             response.nodes = self._route_request(is_refresh, session, params.node_path)
 
             request_context.send_notification(EXPAND_COMPLETED_METHOD, response)
-        except Exception as e:
-            self._expand_node_error(request_context, params, str(e))
+        except BaseException as e:
+            if session.server.connection is not None and session.server.connection.connection.broken and not retry:
```

Comment:
It is great to have safeguard for recursive calls. But it's little confusing since you're retrying when retry is set to False. Do you want to call this method with retry=True and call the recursive call with retry=False? Let me know what you think

Reply:
ohhh or are you using retry parameter to represent this method is being retried?? then that makes more sense..

Reply:
Yup, Using it to track if this is the second attempt. Will rename to help confusion